Sprint-1/exercises/initials.js

```diff
@@ -4,3 +4,17 @@ let lastName = "Johnson";
 
 // Declare a variable called initials that stores the first character of each string.
 // This should produce the string "CKJ", but you must not write the characters C, K, or J in the code of your solution.
+//const initials = firstName[0] + middleName[0] + lastName[0]
+
+// make first, middle, and last name a string and convert it into an array
+const fullNameToArr = (firstName + " " + middleName + " " + lastName).split(' ')
```

Comment:
This is a really interesting approach! It works, but I have some questions! Can you talk through how you came up with this solution? Can you explain a few other ways you could have solved this problem, and why this is the one you chose?

Reply:
Ohh yeah I knew I could have just gotten the first character of each string(firstName[0] , middleName[0], lastName[0]) using the their indeces and concatenate them as follows: 
```const initials = firstName[0] + middleName[0] + lastName[0]```
This is actually what I did in the first place but I want to explore more ways to do it. That's when I came up with the idea of using an array. 

This line here ```const fullNameToArr = (firstName + " " + middleName + " " + lastName).split(' ')``` concatenate the 3 strings and use the split method to convert them into an array. 

And in here ```for (let key in fullNameToArr){
    const value = fullNameToArr[key]
    initials += value[0]
}``` I am using a for in loop to get the values of the array which are going to be : Creola, Katherine and Johnson. and finally get the initial character of each value

Any other way I could have done it better?

Reply:
Sounds good!

You'll learn more about arrays later, but one other thing you could've done is created the array directly rather than by concatenating the strings and splitting them, i.e. `const namesArray = [firstName, middleName, lastName];` - this is a bit simpler and quicker than making a string and splitting it. While it would work exactly the same in _this_ example, it does not actually do exactly the same thing. Can you think what cases would be handled _differently_ by the two approaches?

Also, note how we used different variable names - you used `fullNameToArr` and I used `namesArray` (or honestly probably just `names`). Can you think why I prefer my variable name to yours? :)

Reply:
Thank your for sharing this approach. 

I think in yours, we do not need to add the seperator the array is automatically created while in mine the seperator had to be added while concatinating the names for the split method to work as intended. 

Also we would use the ```split``` method if we wanted to create a new array so that a change in the new array will not affect the original array.

I am not sure why you prefer your variable name instead of mine. Maybe because it is more meaningful ?

Reply:
In general variable names should represent _what_ is in the variable, not _how_ it got there.

Imagine you're making a website that allows you to log in by email or by Facebook. In most of the code of your website, you would want to be able to get the ID of the user. But in most places, it doesn't matter how they logged in. You'd call the variable `userId` not `userIdFromFacebook`.

(This isn't _always_ true - in your code that specifically handles log-in, may be both a `userIdFromFacebook` variable and a `userIdFromEmail` variable specifically so you know the difference between them. But if you only have _one_ variable representing some data, you probably want a really general name describing what the data is).